@viewsort/plot.m

```diff
@@ -62,43 +63,83 @@
-		sdstr = get(obj,'SessionDirs');
-		title(getDataOrder('ShortName','DirString',sdstr{n}))
-	end  % if(Args.Array || Args.Session || Args.Day)
+        xind = (obj.data.ChannelIndex(n)+1):obj.data.ChannelIndex(n+1);
+        plot((obj.data.spikeForms(xind,:))','.-')
+        sdstr = get(obj,'SessionDirs');
+        title(getDataOrder('ShortName','DirString',sdstr{n}))
+        
+        % plot noise on top of waveform
+        hold on
+        line(repmat(xlim',1,2),repmat([-obj.data.Noise(n) obj.data.Noise(n)],2,1),'Color','r')
+        % plot(obj.data.Noise(xind(end),:),'r')
+        % plot(0-(obj.data.Noise(xind(end),:)),'r')
+        hold off
+        
+        % add axis labels
+        if(~Args.LabelsOff)
+            xlabel('Data Points')
+            ylabel('Voltage (uV)')
+        end
+        
+        % show ISI coefficients of variation
+        legendLabels = cell(1,size(xind,1));
+        for k = 1:size(xind,2)
+            legendLabels{k} = num2str(round(obj.data.coeffV_ISI(xind(k)),2));
+        end
+        lgd = legend(legendLabels, 'FontSize', 12);
+        title(lgd,{'ISI Distribution','Coefficient of Variation: '})
+        % subplot(round((numUnits+1)/2),2,d+1);
+        % histogram(spike_ISI,1000,'facecolor',plotColour(d),'edgecolor',plotColour(d)); hold on
+        % title(strcat('(',num2str(d),')',' ISI Distribution, Coefficient of Variation: ', num2str(round(coeffV_ISI,2))));
+        % xlabel('ISI(ms)','FontSize',10); ylabel('Count','FontSize',10);
+       
+    end  % if(Args.Array || Args.Session || Args.Day)
 else
 	% plot all data
 	numSets = obj.data.numSets;
     for index = 1:numSets
         if(numSets>1)
             nptSubplot(numSets,index);
         end
-		xind = (obj.data.ChannelIndex(index)+1):obj.data.ChannelIndex(index+1);
-		plot((obj.data.spikeForms(xind,:))','.-')
-		sdstr = get(obj,'SessionDirs');
-		title(getDataOrder('ShortName','DirString',sdstr{index}))
-        
-        % add code for plot options here
-        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-        %
-        % @viewsort/PLOT takes 'LabelsOff' as an example
+        xind = (obj.data.ChannelIndex(index)+1):obj.data.ChannelIndex(index+1);
+        plot((obj.data.spikeForms(xind,:))','.-')
+        sdstr = get(obj,'SessionDirs');
+        title(getDataOrder('ShortName','DirString',sdstr{index}))
+        % plot noise on top of waveform
+        hold on
+        line(repmat(xlim',1,2),repmat([-obj.data.Noise(xind) obj.data.Noise(xind)],2,1),'Color','r')
+        % plot(obj.data.Noise(xind(end),:),'r')
+        % plot(0-(obj.data.Noise(xind(end),:)),'r')
+        hold off
+        % add axis labels
         if(~Args.LabelsOff)
             xlabel('Data Points')
             ylabel('Voltage (uV)')
         end
-        %
-        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-	end
+    end
+end
+
+% show spike similarities (dot product)
+
+
+
+if(~isempty(Args.Cmds))
+    cd(sdstr{n})
```

Comment:
We might need to save the handle to the current figure before evaluating Args.Cmds so we can set the current figure back to the original figure after running Args.Cmds. You can take a look at the code in rplhighpass/plot.m for an example.